server/etcdserver/api/etcdhttp/health.go

```diff
@@ -99,30 +103,53 @@ func NewHealthHandler(lg *zap.Logger, hfunc func(ctx context.Context, excludedAl
 }
 
 var (
+	// Deprecated: use healthCheckCounter instead.
 	healthSuccess = prometheus.NewCounter(prometheus.CounterOpts{
 		Namespace: "etcd",
 		Subsystem: "server",
 		Name:      "health_success",
-		Help:      "The total number of successful health checks",
+		Help:      "The total number of successful health checks (Deprecated, use etcd_server_healthchecks_total instead)",
 	})
+	// Deprecated: use healthCheckCounter instead.
 	healthFailed = prometheus.NewCounter(prometheus.CounterOpts{
 		Namespace: "etcd",
 		Subsystem: "server",
 		Name:      "health_failures",
-		Help:      "The total number of failed health checks",
+		Help:      "The total number of failed health checks (Deprecated, use etcd_server_healthchecks_total instead)",
 	})
+	healthCheckGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "etcd",
+		Subsystem: "server",
+		Name:      "healthcheck",
+		Help:      "This metric records the result of a single healthcheck.",
+	},
+		[]string{"type", "name"},
+	)
+	healthCheckCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "etcd",
+		Subsystem: "server",
+		Name:      "healthchecks_total",
+		Help:      "This metric records the total result counts of all healthchecks.",
+	},
+		[]string{"type", "name", "status"},
+	)
 )
 
 func init() {
 	prometheus.MustRegister(healthSuccess)
 	prometheus.MustRegister(healthFailed)
+	prometheus.MustRegister(healthCheckGauge)
+	prometheus.MustRegister(healthCheckCounter)
 }
 
 // Health defines etcd server health status.
 // TODO: remove manual parsing in etcdctl cluster-health
 type Health struct {
+	// Health field is used in old /health check path.
 	Health string `json:"health"`
 	Reason string `json:"reason"`
+	// Status field is used in new /readyz or /livez health checks instead of the Health field.
+	Status string `json:"-"`
```

Comment:
Did you intentionally use "-"?
```suggestion
	Status string `json:"status"`
```

Reply:
This is ignored because I don't want to change the response of existing /health endpoint. If use `json:"status"`, the response would add an empty status field.

Reply:
Why introduce this field `Status`?  Why not to reuse the existing `Health` to check whether the health check is successful or failed?  If you are planning to remove the fields "Health" and "Reason" and the legacy health check endpoint /health in future release e.g. 4.0, please add a comment.

Reply:
We still would like to use the `Health` struct. But the `Health` string field is itself not very descriptive and not using constant strings. Added a comment about deprecation.

Reply:
Can we have separate `Health` struct for new endpoints?
Old will use `Health` and `Reason` fields, while new will use `Reason` and `Status`.

Reply:
> Old will use `Health` and `Reason` fields, while new will use `Reason` and `Status`.

Sounds good to me.

Reply:
Added a new `HealthStatus` struct.